### ISSUE TITLE
The Climb doesn't run

### ISSUE THREAD

**Issue by stuff13** [LibreVR/Revive#118]
Lucky's and a few others run fine for me using Revive 0.6.
When I try to start The Climb, I get a dialog that says:
"A connection to the Oculus Store could not be established. Please check your internet connection 
and firewall and make sure that you are logged into your Oculus Store account."

Anybody had any luck with Revive 0.6?
Am I supposed to be running the Oculus client while I use Revive? (I've tried on and off)
Am I supposed to use the Injector for it? (I've tried that)
Any suggestions about what I may be doing wrong?

Thanks for any assistance...

**Comment by LibreVR** [comment]
My response from the Reddit thread:

> The Climb uses Denuvo Anti-Tampering, which wasn't a problem when the headset check was not part of the DRM.

> Seeing as The Climb doesn't have any Vive support announced yet, I may take a swing at Denuvo sometime. Can't make any promises though, that is a serious DRM protection.

> But luckily I'm not interested in actually pirating the game, I just want to fool it into thinking the headset is connected. So I may eventually be able to find another way around it that doesn't trigger the DRM.

**Comment by stuff13** [comment]
Well, I look forward to when you can get a chance to get it working.
If you need any help from me, let me know...

Hope all is well,
Michael M.

**Comment by DragonQuix** [comment]
I encountered the same problem, unfortunately. How can I downgrade to Oculus Runtime 1.3 and download an older version of Revive (0.5)?

**Comment by DragonQuix** [comment]
Do some searching and found this post: https://www.reddit.com/r/Vive/comments/4keupy/downgrading_to_oculus_13/

I'd give it a try, thanks!

**Comment by Torsten85** [comment]
I still get the same error message mentioned above after downgrading oculus to 1.3.2 and revive to 0.5. Did anyone manage to get the climb running?

**Comment by DragonQuix** [comment]
Hi. I just managed to get it to work. Install revive 0.5.2 instead of 0.5. Good luck!

**Comment by LibreVR** [comment]
@DragonQuix Since you got it to work with Oculus Runtime 1.3 could you also test it out with Revive 0.6? Then I can confirm that the DRM patch is triggering the Denuvo protection.

**Comment by DragonQuix** [comment]
Actually I tried 0.6 first and it did not work. I was able to start the game (in the headset only, the monitor of my computer was blacked out, which is not the case when I start the game with 0.5.2 ). But every time I got to the point when the game show "loading..." on the bottom left of the square screen showed in the headset and the next thing it showed was just the streamVR start screen. I could press the home button on the vive controller and it showed that the game was running. The computer monitor was still blacked out. And if  I tried to press the keyboard the windows would tell me the game stopped responding. And I was not able to get pass that point no matter what I did.

I hoped that helps. I just finished the first climb of the valley. So I can report that 0.5.2 combined with downgraded oculus home 1.3 right now totally works for playing The Climb. Thank you for your wonderful software! I loved it!

**Comment by LibreVR** [comment]
Hmm, that still doesn't really tell me whether the DRM is being triggered.

**Comment by DragonQuix** [comment]
So what can I do to help?

**Comment by peatymasta** [comment]
I've managed to downgrade my runtime to 1.3 (thank you for the guide on that, by the way). 

Where can I download the 0.5 version of ReVive?

**Comment by DragonQuix** [comment]
You are welcome. I like to help. You can download the old versions by go to top left area of this page and click "code". Then on the page that shows up you'll see a color bar. Just above that bar find and click the "releases" link. Scroll down to find the version you want. Good luck!

**Comment by Torsten85** [comment]
@DragonQuix Did you also block the inbound and outbound connections for oculus? 
I still get the same error as mentioned above. I've installed 0.5.2 of revive and also 1.3.2 of oculus.
What exactly is your startup procedure? 

Thanks.

**Comment by DragonQuix** [comment]
Hi. I downgraded oc home to 1.3 by using the patch and following the instrutions in this post : https://www.reddit.com/r/Vive/comments/4kd08m/revive_compatible_oculus_runtime_132_is_still_on
Did you do it the same way?

 I don't block the connections now. I did block them at first but after careful tests I did not find any differences either way. I kept oc home online for hours to download new games and everything worked just fine. When I wanna play The Climb I installed 0.5.2. And when I played Chronos and other games that can not run with 0.5.2, I just installed 0.6 and everything worked fine. So now by switching back and forth 0.5.2 and 0.6 I get to play two of my favorite games: The Climb and Chronos. Revive is amazing!

**Comment by supermanscottbristol** [comment]
So I've done the same as you, downgraded 1.4 to 1.3 using the patch.
For some reason The Climb doesn't show up in my Revive tab, so I've tried using the injector, both 0.5.2 and 0.6,  neither work, both with the firewall enabled and disabled.

How are you starting The Climb up ?  Via Revive tab in Steam or injector or something else?

**Comment by DragonQuix** [comment]
I just launch SteamVR directly and in it call up the Revive tab and I can see The Climb, click it and the game would begin.

I think you may have overwritten your game installation data of oc home in the downgrade process. Did you try launching the oc home program to check if you have The Climb in your library? My guess is that if it doesn't show up in your oc home library. It won't show up in the revive tab. In that case. You may have to redownload the game （or overwrite your Software and Manifests folders if you did make a copy before you downgraded） .

**Comment by supermanscottbristol** [comment]
Looks like you're right.  I think I deleted my Manifest for The Climb.  I've still got the game in the software section.
Any chance you could zip me up your manifest section for The Climb please?  It's only a few k.  I'm way out in the country on a 2meg line, took me 3 days to download the game last time, I really don't want to have to do that for just a couple of k.

**Comment by DragonQuix** [comment]
Sure. But I can't do it right now. Maybe in 1-2 hours. Do you have an e-mail address?

**Comment by supermanscottbristol** [comment]
ahh man you're a legend.  Hope that sorts it.   Thanks ever so much.
email : scott.hillier@blueyonder.co.uk

**Comment by DragonQuix** [comment]
Sent. Check your mail and, good luck!

**Comment by stuff13** [comment]
I have The Climb showing up in my Oculus Library with the option to Start.
I reverted to 5.2, I downgraded my Oculus library...
When I start The Climb in Revive, I just get the dialog saying that I can't
connect to the store.
So, there must be some steps missing somewhere.

Any thoughts?

Hope all is well,
Michael MacKinnon

**Comment by supermanscottbristol** [comment]
So I'm in the same position as Michael, only I somehow must have screwed my library when I downgraded as The Climb no longer appears.  
After adding the missing Manifest files DragonQuix was good enough to send me, I do now have an extra square in Revive (which I think must represent The Climb but without the jpg) clicking it still doesn't start the game.  When I use the injector I still get the same message as Michael.  meh

**Comment by stuff13** [comment]
I just found some Revive Injector files in my Climb folder. 
Revive Injector changes the exe doesn’t in? We could be failing the Denovo check.
We may need to delete the changed files and redownload the files again through the Oculus Store. 

And NOT user the Injector on them.

Maybe?

Hope all is well,
Michael MacKinnon

**Comment by supermanscottbristol** [comment]
Tried that now.  Id backed the software folder up before I started fiddling.  So I put The Climb back in place and it's no different.  Still get the message about not being able to contact the store.
I checked the ReviveInjector log file and it just says cannot create process.

**Comment by stuff13** [comment]
Yeah. 
Deleted all the files in the climb folder. 
Uninstalled in Oculus store.
Replaced the 1.40 Oculus server files
restart oculus server
Installed The Climb from the oculus store
Installed the 0.5.2 version of revive
shut down oculus server
copied over the 1.32 oculus server files
restart oculus server
restart steamvr
go into steamvr
go to the revive tab
select The Climb

It doesn’t start and all I get for Christmas is the dialog telling me that it can’t connect to the oculus store.

sigh.
In the list above, am I missing anything?
Should I be using the 0.5 version of Revive? Should I be deleting some saved files somewhere?
Juggling eggs on the second Tuesday in June?

Hope all is well,
Michael M

**Comment by wissper** [comment]
Sounds like they added a server side response of some kind...  If you could figure out the IP address / host name maybe creating a dns entry in your hosts file would satisfy the ping to the server.

Sent from my T-Mobile 4G LTE Device
-------- Original message --------From: Michael MacKinnon <notifications@github.com> Date: 6/1/16  1:18 PM  (GMT-05:00) To: LibreVR/Revive <Revive@noreply.github.com> Subject: Re: [LibreVR/Revive] The Climb doesn't run (#118) 
Yeah. 

Deleted all the files in the climb folder. 

Uninstalled in Oculus store.

Replaced the 1.40 Oculus server files

restart oculus server

Installed The Climb from the oculus store

Installed the 0.5.2 version of revive

shut down oculus server

copied over the 1.32 oculus server files

restart oculus server

restart steamvr

go into steamvr

go to the revive tab

select The Climb



It doesn’t start and all I get for Christmas is the dialog telling me that it can’t connect to the oculus store.



sigh.

In the list above, am I missing anything?

Should I be using the 0.5 version of Revive? Should I be deleting some saved files somewhere?

Juggling eggs on the second Tuesday in June?



Hope all is well,

Michael M

**Comment by DragonQuix** [comment]
Hey guys. Sorry to hear that. After reading the list stuff13 wrote in the above post. Here are the things I would do differently: I'd install the 0.5.2 version of revive after I finished copying the 1.32 oc runtime folder, instead of before that. Second thing is I am not sure if "shut down oculus server" in that list means blocking the outbound connections from the oc programs? If yes then that's what I'd do. Also you should know that there are at least three programs needed to be blocked. Check this out: http://puu.sh/oZP5W/f2d69dfeae.png .Lastly, when I finished installing the 0.5.2 revive. I did not unblock the programs. Everything worked fine for me, except the game would tell me can not retreat world record other gamers created or something like that. So there's really no need to unblock the programs to play the game.

**Comment by supermanscottbristol** [comment]
No joy after doing all that still.
What is the status of your 2 Oculus services ?
Do you have the Oculus Home software running when you start The Climb up ?
Do you have blocks for inbound, or just outbound?

thanks.

**Comment by DragonQuix** [comment]
I am not sure about the services status. I did check once or twice. One of them was set to auto start, and it's  status was start. The other was not. But overall I just leave them be. 
No. The Oculus Home was not running most of the time when I launch streamVR and load the game in the revive tab. Sometimes I left it runing just because I was downloading some other game at the same time. But that situation is rare.
I use a software called proxifier to block the programs. And my blocking list are more extendsive than just the three. But since peatymasta in this post :https://www.reddit.com/r/Vive/comments/4lqfsc/has_anyone_gotten_the_climb_working_with_revive/ said that he/she also succeed in getting the game to run. I don't think that makes any differences. But I'll share my blocking list just to let you guys be sure, here they are: OculusSetup.exe;OculusClient.exe;OVRServiceLauncher.exe;OVRServer_x64.exe;OculusVR.exe;OculusCompatCheck.exe;CompatToolCommandLine.exe;OculusLogGatherer.exe;OVRLibrarian.exe;oculus-driver.exe;OVRLibraryService.exe;oculus-overlays.exe;OVRRedistributableInstaller.exe

**Comment by stuff13** [comment]
Another point that might be relevant: I'm running on Windows 10
(Well, 'not running' is more accurate)


Mike M

**Comment by DragonQuix** [comment]
@stuff13 I am running on windows 10, also. So I guess you can uncheck that.

**Comment by Torsten85** [comment]
I got it working by simple disabling the rules in the firewall. The I could simply start the climb from the revive tab without having oculus client open.

@DragonQuix: Did you block inbound and outbound communication or only one of them?

**Comment by supermanscottbristol** [comment]
I'm starting to wonder if you don't have it right the first time you try it, something is written into your files which prevents it from running somehow ?  I've tried everything everyone has suggested in here and always end up with the Store cant be contacted message.

**Comment by DragonQuix** [comment]
@supermanscottbristol I don't think that's the case. I think you just need to redo every step from the begining, and do it carefully. If that didn't work, think about something you did not do and do it again. That's how I usually solve this kind of problems.

**Comment by scotthillier** [comment]
As I said above I have redone every possible step in just about every possible combination. Something is different but without any kind of pointer from the revive log it's hard to know what could be different

**Comment by supermanscottbristol** [comment]
As I said above I have redone every possible step in just about every possible combination. Something is different but without any kind of pointer from the revive log it's hard to know what could be different

**Comment by LibreVR** [comment]
If you post your logs I can take a look at them. But I don't actively support downgrading the Oculus Runtime, it's very error-prone. I hope to fix The Climb when I finally find a good way to make Revive work without bypassing the DRM.

**Comment by supermanscottbristol** [comment]
@LibreVR  happy to.  I'll upload the 2 Revive ones using version 0.5.2 and version 0.6.  If there are any others you need, let me know.  cheers!

**Comment by supermanscottbristol** [comment]
Ok, so I've 6 files here.  A 0.6 and 0.5.2 overlay log.  Then a revive injector log for both versions, 1 for launching from inside revive - which actually doesn't result in the error message, just nothing happens, the other is for when I drag the climb exe onto the injector.

When I installed 6 I did it to C:\ and for 52 I did it to V:\ just so I could easily differentiate between log files.  I never had both versions installed at once.

Couple of things I've noticed -  when you do the drag method for the first time Windows Smart Screen gets in the way and you have to tell it to allow the file to run.  Once you've done it once it then doesn't ask again.  I wondered if maybe that's what was happening behind the scenes when you try and launch from in VR so I tried running the injector file from in the revive install directory just in case smart screen would pop up but that didn't change anything.

When uninstalling 6 and going to 52 and vice versa, the Steam VR would complain about the compositor not running and prevent the headset from working until I rebooted.
After a reboot - all good.

**Comment by LibreVR** [comment]
Seems like the injector is fine, but the dashboard is detecting an incorrect folder for The Climb, so the dashboard won't be able to start the application.

**Comment by scotthillier** [comment]
Really? Where can you see anything about incorrect folders mate??

**Comment by supermanscottbristol** [comment]
Ah ok. Dunno how I've screwer that up then. Thanks for looking. Maybe in the future if there's a debug mode I might be able to find out what I've done. 
Cheers

**Comment by peatymasta** [comment]
I've got another issue. I was able to get the Climb working, but now it's launching in full screen (and Steam VR is giving me the "compositor must be full screen" error. When I try to click it on the Windows 10 dock to possibly minimize it, screen focus just changes to the game. I can't use the HMD to select any of the windowed options...
Is there any way to force open it in windowed mode? I know I can use the command -window in a shortcut, but it seems that i need to open the .exe with ReVive

**Comment by LibreVR** [comment]
@peatymasta You should not be using extended mode with Revive, try to enable Direct Mode in SteamVR.

**Comment by peatymasta** [comment]
That did the trick! Thanks so much. Do you guys have a donation page?

**Comment by LibreVR** [comment]
Nope, I usually tell people to donate to the [EFF](https://supporters.eff.org/donate) instead.

**Comment by supermanscottbristol** [comment]
@stuff13 Did you ever get it going ?

**Comment by stuff13** [comment]
Nope.
No success from me.

Michael M

**Comment by supermanscottbristol** [comment]
Could someone give me a list of the directory structure of their Climb directories please?  LibreVR thinks I might have issues with my folders but I cant see how.

Thanks a lot

**Comment by LibreVR** [comment]
Fixed now that the headset check was moved out of the DRM.

**Comment by supermanscottbristol** [comment]
What is needed? A new version of home or a new version of the climb??

**Comment by LibreVR** [comment]
You need to update to the latest version of Revive and of Oculus Home and you need to use the Injector instead of the Dashboard.
